app_integrations/apps/salesforce.py

```diff
@@ -0,0 +1,395 @@
+    _SALESFORCE_QUERY_URL = ('{instance_url}/services/data/v{api_version}/'
+                             '{query}{start_time}{event_type}')
+    # Use the Query resource to retrieve log files.
+    _SALESFORCE_QUERY_FILTERS = ('query?q=SELECT+Id+,+EventType+,+LogFile+,+LogDate+,'
+                                 '+LogFileLength+FROM+EventLogFile+')
+
+    # Log files filters. These typs of events should be auditted for security purpose.
+    _SALESFORCE_EVENT_TYPES = {'Console', 'Login', 'LoginAs', 'Report', 'ReportExport'}
+    # Use "Where" and "LogDate" clause to filter log files which are generated after
+    # last_timestamp.
+    _SALESFORCE_CREATE_AFTER = 'WHERE+LogDate+>+{}+'
+
+    _TIMEOUT = 15
+    EXCEPTIONS_TO_BACKOFF = (SalesforceAppError,)
+    BACKOFF_MAX_RETRIES = 3
+
+    def __init__(self, config):
+        super(SalesforceApp, self).__init__(config)
+        self._auth_headers = None
+        self._instance_url = None
+        self._latest_api_version = 0
+
+    @classmethod
+    def _type(cls):
+        return 'events'
+
+    @classmethod
+    def service(cls):
+        return 'salesforce'
+
+    @classmethod
+    def date_formatter(cls):
+        """Salesforce API date format: YYYY-MM-DDTHH:MM:SSZ"""
+        return '%Y-%m-%dT%H:%M:%SZ'
+
+    def _request_token(self):
+        """Request OAuth token frome salesforce
+
+        Meanwhile, it will also get instance url which will be used in futher
+        requests. The instance url identifies the Salesforce instance to which
+        API calls should be sent.
+
+        Returns:
+            bool: Returns True if update auth headers and instance url successfully.
+        """
+        headers = {'Content-Type': 'application/x-www-form-urlencoded'}
+        # required credentials when request for a token.
+        data = {
+            'grant_type': 'password',
+            'client_id': self._config.auth['client_id'],
+            'client_secret': self._config.auth['client_secret'],
+            'username': self._config.auth['username'],
+            'password': self._config.auth['password']+self._config.auth['security_token'],
+            'response_type': 'code',
+            'redirect_uri': self._SALESFORCE_TOKEN_URL
+        }
+        success, response = self._make_post_request(self._SALESFORCE_TOKEN_URL, headers, data)
+
+        if not success:
+            return False
+
+        if not response or not response.get('access_token') or not response.get('instance_url'):
+            LOGGER.error('Response invalid generating headers for service \'%s\'',
+                         self.type())
+            return False
+
+        bearer = 'Bearer {}'.format(response.get('access_token'))
+        self._auth_headers = {
+            'Content-Type': 'application/json',
+            'Authorization': bearer
+        }
+        self._instance_url = response.get('instance_url')
+        LOGGER.debug('Successfully obtain OAuth token and instance URL')
+        return True
```

Comment:
consider having this return value be `return self._instance_url and self._auth_headers` (would be `False` if either of those are `None`)